### DISCUSSION TITLE
Activity function overloading

### DISCUSSION THREAD

**Discussion by yell0wfl4sh** [Azure/azure-functions-durable-extension#2288] [Q&A] [answered]
We are noticing intermittent failures with overloaded activities, even if these activities are in different orchestrators. In our implementation, we have two activities with same function name and they take different input object type. 

Error log: 
```
D:\dbs\el\dbc1\sources\dev\Workflow\Core\TrainingCampaignV2\TrainingCampaignV2Orchestrator.cs(519,183): error DF0108: Activity function named 'GetTrainingEventContentActivity' takes a parameter of type 'Microsoft.AttackSimulator.ObjectModel.Core.SimulationLaunchMessage' but was given 'Microsoft.AttackSimulator.ObjectModel.Core.TrainingCampaigns.TrainingCampaignV2LaunchMessage'. [D:\dbs\el\dbc1\sources\dev\Workflow\Core\Microsoft.AttackSimulator.Workflow.Core.csproj]
```

I am suspecting this might be caused because of implementation where in our orchestration code, we call activities this way:
`await context.CallActivityWithRetryAsync<TrainingContent>(nameof(GetTrainingEventContentActivity), this.retryOptions.General, launchMessage).ConfigureAwait(true);` 
instead of:
`await context.CallActivityWithRetryAsync<TrainingContent>(nameof(this.GetTrainingEventContentActivity), this.retryOptions.General, launchMessage).ConfigureAwait(true);`

Please help us confirming this theory!

Thanks,
Yash

**Comment by cgillum** [comment, marked as answer]
AFAIK Azure Functions doesn't support multiple functions with the same name, regardless of whether they are activity functions or something else. That makes me think that there might be some misunderstanding about what's actually happening.

If I'm reading this correctly, the difference between those two lines you shared is `nameof(GetTrainingEventContentActivity)` vs. `nameof(this.GetTrainingEventContentActivity)`. However, both expressions should get compiled down to "GetTrainingEventContentActivity". You can see an example here: https://dotnetfiddle.net/IAFXto.

**Reply by yell0wfl4sh** [reply]
Thanks for confirming @cgillum! We had implemented activities with same name and different arguments across orchestrators assuming overloading would work fine. This error was reported in runtime, will adding an analyzer to make sure this gives build error be a good addition?

**Reply by cgillum** [reply]
Yeah, I think an analyzer for this would be ideal. It's probably best to open a GitHub issue for the Functions runtime team to look at. @fabiocav, where do you think a good repo to open such an issue would be?
